### ISSUE TITLE
Wrong error when `derive(CoercePointee)` is used with `cfg_attr`'ed `pointee` and 2+ generic argument type

### ISSUE THREAD

**Issue by WaffleLapkin** [rust-lang/rust-analyzer#20500]
**rust-analyzer version**: `rust-analyzer 1.91.0-nightly (425a9c0 2025-08-17)`

**rustc version**: `rustc 1.91.0-nightly (425a9c0a0 2025-08-17)`

**editor or extension**: helix

**code snippet to reproduce**:
```rust
#![feature(derive_coerce_pointee)]
use std::marker::CoercePointee;
use std::ptr::NonNull;

#[derive(CoercePointee)]
#[repr(transparent)]
pub struct Ptr<T, #[cfg_attr(all(), pointee)] V: ?Sized> {
    ptr: NonNull<V>,
    _ghost: PhantomData<T>,
}
```
r-a emits an error saying "exactly one generic type parameter must be marked as `#[pointee]` to derive `CoercePointee`", rustc compiles this code successfully.

**Comment by ChayimFriedman2** [comment]
Probably will be fixed by https://github.com/rust-lang/rust-analyzer/pull/20316, which refactors how we handle `cfg` and `cfg_attr` passed to macros. But it will have to wait until we complete the migration to the next solver.

**Comment by ChayimFriedman2** [comment]
This is indeed fixed now.
